x-pack/legacy/plugins/reporting/export_types/common/lib/screenshots/observable.ts

```diff
@@ -36,13 +36,16 @@ export function screenshotsObservableFactory(
       { viewport: layout.getBrowserViewport(), browserTimezone },
       logger
     );
-    return Rx.from(urls).pipe(
-      concatMap(url => {
-        return create$.pipe(
-          mergeMap(({ driver, exit$ }) => {
+
+    return create$.pipe(
+      mergeMap(({ driver, exit$ }) => {
+        return Rx.from(urls).pipe(
+          concatMap((url, index) => {
             const setup$: Rx.Observable<ScreenSetupData> = Rx.of(1).pipe(
               takeUntil(exit$),
-              mergeMap(() => openUrl(captureConfig, driver, url, conditionalHeaders, logger)),
+              mergeMap(() =>
+                openUrl(captureConfig, driver, url, conditionalHeaders, logger, index + 1)
```

Comment:
I'd prefer `logger` to stay as the final argument. I find that keeping all these functions that way makes it easier to read them.

Reply:
👍